core/src/main/scala/cats/Eval.scala

```diff
@@ -246,56 +246,60 @@ object Eval extends EvalInstances {
   val One: Eval[Int] = Now(1)
 
   /**
-   * Call is a type of Eval[A] that is used to defer computations
+   * Defer is a type of Eval[A] that is used to defer computations
    * which produce Eval[A].
    *
-   * Users should not instantiate Call instances themselves. Instead,
+   * Users should not instantiate Defer instances themselves. Instead,
    * they will be automatically created when needed.
    */
-  sealed abstract class Call[A](val thunk: () => Eval[A]) extends Eval[A] {
+  sealed abstract class Defer[A](val thunk: () => Eval[A]) extends Eval[A] {
 
     def memoize: Eval[A] = Memoize(this)
     def value: A = evaluate(this)
   }
 
   /**
-   * Collapse the call stack for eager evaluations.
-   * returns a non Call Eval node
+   * Advance until we find a non-deferred Eval node.
+   *
+   * Often we may have deep chains of Defer nodes; the goal here is to
+   * advance through those to find the underlying "work" (in the case
+   * of FlatMap nodes) or "value" (in the case of Now, Later, or
+   * Always nodes).
    */
-  @tailrec private def doCall[A](fa: Eval[A]): Eval[A] =
+  @tailrec private def advance[A](fa: Eval[A]): Eval[A] =
     fa match {
-      case call: Eval.Call[A] =>
-        doCall(call.thunk())
-      case compute: Eval.Compute[A] =>
-        new Eval.Compute[A] {
+      case call: Eval.Defer[A] =>
+        advance(call.thunk())
+      case compute: Eval.FlatMap[A] =>
+        new Eval.FlatMap[A] {
           type Start = compute.Start
           val start: () => Eval[Start] = () => compute.start()
-          val run: Start => Eval[A] = s => doCall1(compute.run(s))
+          val run: Start => Eval[A] = s => advance1(compute.run(s))
```

Comment:
Why do we need this here? Seems like it will be run later without beating the function call here.